From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 30 Jul 2016 09:56:50 +0200
Subject: [PATCH] {MAYBE UPS} zram: make clang happy about strlcpy usage

clang complains:

    drivers/block/zram/zram_drv.c:364:47: error: size argument in
      'strlcpy' call appears to be size of the source; expected the size of the destination
      [-Werror,-Wstrlcpy-strlcat-size]
        strlcpy(zram->compressor, compressor, sizeof(compressor));
                                              ~~~~~~~^~~~~~~~~~~
    drivers/block/zram/zram_drv.c:364:40: note: change size
      argument to be the size of the destination
        strlcpy(zram->compressor, compressor, sizeof(compressor));
                                              ^~~~~~~~~~~~~~~~~~
                                              sizeof(zram->compressor)

even thoung sizeof(compressor) == sizeof(zram->compressor) ==
CRYPTO_MAX_ALG_NAME.  Add a BUILD_BUG_ON to check this property and make
clang happy.
---
 drivers/block/zram/zram_drv.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/block/zram/zram_drv.c b/drivers/block/zram/zram_drv.c
index 0c09d4256108..b5dfa0296d73 100644
--- a/drivers/block/zram/zram_drv.c
+++ b/drivers/block/zram/zram_drv.c
@@ -320,7 +320,8 @@ static ssize_t comp_algorithm_store(struct device *dev,
 		return -EBUSY;
 	}
 
-	strlcpy(zram->compressor, compressor, sizeof(compressor));
+	BUILD_BUG_ON(sizeof(zram->compressor) != sizeof(compressor));
+	strlcpy(zram->compressor, compressor, sizeof(zram->compressor));
 	up_write(&zram->init_lock);
 	return len;
 }
-- 
